### PR TITLE
Slider rebase work

### PR DESCRIPTION
**Pull request by JoseAlba**
## Description

Update the Slider and RangeSlider to the latest Material spec. This introduces an updated track, updated thumbs, updated tick marks, and a new value indicator shape. The old paddle value indicator shape is also updated so that text scaling works consistently.

Full details of this change can be found here: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/flutter-announce/69dmlKUL5Ew/tQh-ajiEAAAJ

## Tests

I added the following tests:

Updated all affected tests in slider_test, range_slider_test, and slider_theme_test
Also kept the original paddle value indicator tests and added a new test for the new default rectangular shape.

## Checklist

Update the Slider and RangeSlider to the latest Material spec. This introduces an updated track, updated thumbs, updated tick marks, and a new value indicator shape. The old paddle value indicator shape is also updated so that text scaling works consistently.

Full details of this change can be found here: https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/flutter-announce/69dmlKUL5Ew/tQh-ajiEAAAJ

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [ ] No, no existing tests failed, so this is *not* a breaking change.
- [x] Yes, this is a breaking change. *If not, delete the remainder of this section.*
   - [ ] I wrote a design doc: https://flutter.dev/go/template *Replace this with a link to your design doc's short link*
   - [ ] I got input from the developer relations team, specifically from: *Replace with the names of who gave advice*
   - [ ] I wrote a migration guide: *Replace with a link to your migration guide*

Previous PR: https://github.com/flutter/flutter/pull/52663 that landed but had to get reverted due to a small problem

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: dev/integration_tests/flutter_gallery/lib/demo/material/slider_demo.dart [hunk 1]
```diff
@@ -268,7 +273,8 @@ class _SlidersState extends State<_Sliders> {
                   ),
                 ),
               ),
-              Slider.adaptive(
```
Comment (clocksmith) [inline dev/integration_tests/flutter_gallery/lib/demo/material/slider_demo.dart:271]: I think this should still be adaptive

File: packages/flutter/lib/src/material/slider_theme.dart [hunk 2]
```diff
@@ -1636,10 +1636,12 @@ class RectangularSliderTrackShape extends SliderTrackShape with BaseSliderTrackS
...
     );
 
-    final Rect leftTrackSegment = Rect.fromLTRB(trackRect.left + trackRect.height / 2, trackRect.top, thumbCenter.dx, trackRect.bottom);
+    final Size thumbSize = sliderTheme.thumbShape.getPreferredSize(isEnabled, isDiscrete);
+
+    final Rect leftTrackSegment = Rect.fromLTRB(trackRect.left + trackRect.height / 2, trackRect.top, thumbCenter.dx - thumbSize.width / 2, trackRect.bottom);
     if (!leftTrackSegment.isEmpty)
       context.canvas.drawRect(leftTrackSegment, leftTrackPaint);
-    final Rect rightTrackSegment = Rect.fromLTRB(thumbCenter.dx, trackRect.top, trackRect.right, trackRect.bottom);
+    final Rect rightTrackSegment = Rect.fromLTRB(thumbCenter.dx + thumbSize.width / 2, trackRect.top, trackRect.right, trackRect.bottom);
```
Comment (clocksmith) [inline packages/flutter/lib/src/material/slider_theme.dart]: This should do it!

File: dev/integration_tests/flutter_gallery/lib/demo/material/slider_demo.dart (hunk 1 above)
Comment (JoseAlba) [reply dev/integration_tests/flutter_gallery/lib/demo/material/slider_demo.dart:271]: Done

Comment (JoseAlba) [reply dev/integration_tests/flutter_gallery/lib/demo/material/slider_demo.dart:271]: Done

File: packages/flutter/lib/src/material/slider_theme.dart (hunk 2 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/slider_theme.dart]: Done

File: packages/flutter/lib/src/material/slider.dart [hunk 3]
```diff
@@ -31,6 +31,11 @@ import 'theme.dart';
 ///  * [Slider.semanticFormatterCallback], which shows an example use case.
 typedef SemanticFormatterCallback = String Function(double value);
 
+/// [Slider] uses this callback to paint the value indicator on the overlay.
+/// Since the value indicator is painted on the Overlay; this method paints the
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/slider.dart:35]: blank line here

File: packages/flutter/lib/src/material/slider.dart [hunk 4]
```diff
@@ -374,6 +383,19 @@ class Slider extends StatefulWidget {
   /// Ignored if this slider is created with [Slider.adaptive]
   final SemanticFormatterCallback semanticFormatterCallback;
 
+  /// Whether to use the updated Material spec version of the [Slider].
+  /// * The v2 Slider has an updated value indicator that matches the latest specs.
```
Comment (HansMuller) [inline packages/flutter/lib/src/material/slider.dart:387]: blank line here

**Review by HansMuller — APPROVED** [review]
LGTM, since this is essentially the same PR as https://github.com/flutter/flutter/pull/52663.

File: packages/flutter/lib/src/material/slider.dart (hunk 3 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/slider.dart:36]: Done

File: packages/flutter/lib/src/material/slider.dart (hunk 4 above)
Comment (JoseAlba) [reply packages/flutter/lib/src/material/slider.dart:389]: Done

**Pull request merged**